rfcs/0091-establish-tvm-unity-connection.md

```diff
@@ -0,0 +1,382 @@
+            lv1: R.Tensor[(n * m,), "float32"] = R.flatten(lv0)
+            gv0: R.Tensor[lv2, "float32"] = R.exp(lv1)
+            R.output(gv0)
+
+        # A1: Interaction between "graph" and FFI
+        R.call_packed("custom_inplace_update", gv0)
+        return gv0f
+```
+
+Specifically, unity connection enables the following interactions (annotated in the code above):
+
+- A0: Interaction between graph and Tensor Program (TensorIR)
+- A1: Interaction between graph and FFI
+- A2: Interaction between Tensor Program and Hardware primitive (through tensorization).
+
+Enabling these interactions means:
+
+- We can construct these interactions during model ingestion and use them as an interface for customization — e.g. directly call into both topi (TVM’s operator library) and TensorIR (flexible loop level syntax) in graph construction.
+- We can inspect and transform IRModules that contain those elements to quickly bring in new optimizations.
+- We can build optimization passes by transforming the IRModules and analyzing interactions between different types of abstractions(e.g. graph/TIR).
+
+We have been experimenting with some of the efforts and confirmed that such flow is possible through combination of relax, TensorIR, FFI and AutoTensorization. While some of these capabilities were technically possible to do before TVM Unity, it would have required extensive modifications to TVM internals and increased project complexity. Additionally, many capabilities, such as interaction with symbolic shapes and direct calls into topi, require fundamentally thinking that leads to TVM unity.
+
+Following best practices from other projects as they evolve, Relax will, as a new optional component, be isolated in its own namespace to minimize disruption to the current flow and design organic interfaces to interact with the rest of the unity components. Also, we do not plan to change the current default build pipeline (`tvm.build` and `relay.build`), in order to minimize disruption to existing users. 
+
+To help stage the set of improvements while minimizing disruption of the current flow. We propose and introduce unity connection as an optional compilation flow of TVM, diagrammed as:
+
+![Untitled 1|690x276](assets/0091/unity_connection_diagram.png) 
+
+In the proposal, the “default build” (those that run through `tvmc` and `relay.build`) that TVM users know today will remain unchanged. TVM unity connection will be added as an optional vertical component. 
+
+This is a change that is invisible to the current users who use the existing flow as the default build will continue to work as-is. In the meantime, we give our community members (many of whom are excited about TVM unity) abilities to leverage unity connection early as an optional vertical to enable use cases that may not be supported today (such as dynamic shapes, direct topi integration, and cross abstraction optimizations). 
```

Comment:
I think it is not just about the community members, but also the users and developers who are not in the community but interested in checking it out.